harvester_e2e_tests/integration/test_rancher_integration.py

```diff
@@ -0,0 +1,355 @@
+        code, data = rancher_api_client.mgmt_clusters.get(harvester_cluster_name)
+        if data.get('status', {}).get('ready', False):
+            break
+        sleep(5)
+    else:
+        raise AssertionError(
+            f"MgmtCluster {harvester_cluster_name} can't be ready \
+                with {wait_timeout} timed out\n"
+            f"Got error: {code}, {data}"
+        )
+
+    yield data
+
+    rancher_api_client.mgmt_clusters.delete(harvester_cluster_name)
+
+
+@pytest.fixture(scope='class')
+def harvester_cloud_credential(api_client, rancher_api_client,
+                               harvester_mgmt_cluster, unique_name):
+    harvester_kubeconfig = api_client.generate_kubeconfig()
+    code, data = rancher_api_client.cloud_credentials.create(
+        unique_name,
+        harvester_kubeconfig,
+        harvester_mgmt_cluster['status']['clusterName']
+    )
+
+    assert 201 == code, (
+        f"Failed to create cloud credential with error: {code}, {data}"
+    )
+
+    code, data = rancher_api_client.cloud_credentials.get(data['id'])
+    assert 200 == code, (
+        f"Failed to get cloud credential {data['id']} with error: {code}, {data}"
+    )
+
+    yield data
+
+    rancher_api_client.cloud_credentials.delete(data['id'])
+
+
+@pytest.mark.p0
+@pytest.mark.rancher
+class TestRKE:
+    @pytest.mark.rke2
+    def test_create_rke2(self, rancher_api_client, unique_name, harvester_mgmt_cluster,
+                         harvester_cloud_credential, rke2_cluster_name, focal_image, vlan_network,
+                         k8s_version, wait_timeout):
+        cluster_name = harvester_mgmt_cluster['status']['clusterName']
+        code, data = rancher_api_client.kube_configs.create(
+            rke2_cluster_name,
+            cluster_name
+        )
+        assert 200 == code, (
+            f"Failed to create harvester kubeconfig for rke2 with error: {code}, {data}"
+        )
+        assert "" != data, (
+            f"Harvester kubeconfig for rke2 should not be empty: {code}, {data}"
+        )
+
+        kubeconfig = data
+
+        code, data = rancher_api_client.secrets.create(
+            name=unique_name,
+            data={
+                "credential": kubeconfig[1:-1].replace("\\n", "\n")
+            },
+            annotations={
+                "v2prov-secret-authorized-for-cluster": rke2_cluster_name,
+                "v2prov-authorized-secret-deletes-on-cluster-removal": "true"
+            }
+        )
+        assert 201 == code, (
+            f"Failed to create secret with error: {code}, {data}"
+        )
+
+        cloud_provider_config_id = f"{data['metadata']['namespace']}:{data['metadata']['name']}"
+
+        code, data = rancher_api_client.harvester_configs.create(
+            name=unique_name,
+            cpus="2",
+            mems="4",
+            disks="40",
+            image_id=focal_image,
+            network_id=vlan_network['id'],
+            ssh_user="ubuntu",
```

Comment:
ssh user should not fix in here.
I would suggest to update the focal_image fixture to:
```python
yield dict(ssh_user="ubuntu", id=f"{namespace}/{name}")
```

then if we update the image, the **ssh_user** will be update accordingly.